### PR TITLE
feat: add `utils/async/parallel`

### LINKED ISSUE #1811: [RFC]: add `utils/async/parallel`
### Description

This RFC proposes adding `utils/async/parallel` to run a set of asynchronous code in parallel, corresponding to [async.js parallel](https://caolan.github.io/async/v3/docs.html#parallel) and [async.js parallelLimit](https://caolan.github.io/async/v3/docs.html#parallelLimit)

### Related Issues

Related to the project idea https://github.com/stdlib-js/google-summer-of-code/issues/9

### Questions

No.

### Other

The implementation will be inspired by `utils/async/map-values`.

The API interface would be `parallel( functionArray [, options] [, done] )`.

Params:
- functionArray: an array of functions to be executed in parallel. The functions should be async, or else they would run sequentially.
- `options` will accept: 
  - `thisArg`: execution context
  - `limit`: maximum number of pending invocations with default `infinity`. (Will replace the need for a separate function for `parallelLimit`)
- done: a callback function with (err, result) parameters that is called when the execution is finished.
  - err: will be truthy if an error happened, for error-handling.
  - result: will be an array of the return values of each function.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by nightknighto**
Resolves #1811 .

## Description

This pull request adds a new package `utils/async/parallel`. Execute functions in parallel, passing the results of all functions to a final callback.

Implemented as discussed in the issue.

## Related Issues

-   resolves #1811 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by nightknighto** [comment]
What changes are needed here?

File: lib/node_modules/@stdlib/utils/async/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/README.md:1]: The changes in this file can be removed. We handle this through project tooling and automation.

Comment (nightknighto) [reply lib/node_modules/@stdlib/utils/async/README.md:1]: @kgryte Alright reverted it.

**Comment by nightknighto** [comment]
@kgryte @Planeshifter

File: lib/node_modules/@stdlib/utils/async/parallel/lib/limit.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	* Callback to invoke the next function.
+	*
+	* @private
+	*/
+	function next() {
+		var currentIdx;
+		idx += 1;
+		currentIdx = idx;
+		fcns[idx].call( opts.thisArg, resolve );
+		/**
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/lib/limit.js]:
```suggestion
		fcns[ idx ].call( opts.thisArg, resolve );
		
		/**
```

File: lib/node_modules/@stdlib/utils/async/parallel/lib/factory.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	}
+	doneCb = doneCb.bind( opts.thisArg );
+
+	if ( !opts.limit ) {
+		opts.limit = PINF;
+	}
+	return parallel;
+
+	/**
+	* Maps values from one object to a new object having the same keys.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/lib/factory.js]:
```suggestion
	* Executes a set of functions in parallel and returns an array of results.
```

File: lib/node_modules/@stdlib/utils/async/parallel/package.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
+{
+  "name": "@stdlib/utils/async/parallel",
+  "version": "0.0.0",
+  "description": "Parallel execution of functions",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/async/parallel/package.json]:
```suggestion
  "description": "Execute functions in parallel.",
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/*
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts]:
```suggestion
/**
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+};
+
+const done = ( error: Error | null, out: any  ) => {
+	if ( error ) {
+		throw error;
+	}
+	if ( out === void 0 ) {
+		throw new Error( '`out` is missing.' );
+	}
+};
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts]:
```suggestion
/**
* Test function.
*
* @private
* @param next - callback
*/
function foo( next: Function ): void {
	next( null, 'beep' );
}

/**
* Test function.
*
* @private 
* @param next - callback
*/
function bar( next: Function ): void {
	next( null, 'boop' );
}

/**
* Test function.
*
* @private 
* @param error - error argument
* @param out - output array
*/
function done( error: Error | null, out: any  ): void {
	if ( error ) {
		throw error;
	}
	if ( out === void 0 ) {
		throw new Error( '`out` is missing.' );
	}
}
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+
+// TESTS //
+
+// The function returns void...
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts]:
```suggestion
// The function returns undefined...
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+{
+	parallel( 2, done ); // $ExpectError
+	parallel( false, done ); // $ExpectError
+	parallel( true, done ); // $ExpectError
+	parallel( 'abc', done ); // $ExpectError
+	parallel( {}, done ); // $ExpectError
+	parallel( ( x: number ): number => x, done ); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided a second callback argument which is not a function having a supported signature...
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts]:
```suggestion
// The compiler throws an error if the function is provided an invalid callback argument...
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+	parallel( [ foo, bar ], false ); // $ExpectError
+	parallel( [ foo, bar ], true ); // $ExpectError
+	parallel( [ foo, bar ], 'abc' ); // $ExpectError
+	parallel( [ foo, bar ], {} ); // $ExpectError
+	parallel( [ foo, bar ], [] ); // $ExpectError
+	parallel( [ foo, bar ], ( x: number ): number => x ); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided a third callback argument which is not a function having a supported signature...
+{
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts]:
```suggestion

```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+	parallel( [ foo, bar ], {}, 2 ); // $ExpectError
+	parallel( [ foo, bar ], {}, false ); // $ExpectError
+	parallel( [ foo, bar ], {}, true ); // $ExpectError
+	parallel( [ foo, bar ], {}, 'abc' ); // $ExpectError
+	parallel( [ foo, bar ], {}, {} ); // $ExpectError
+	parallel( [ foo, bar ], {}, [] ); // $ExpectError
+	parallel( [ foo, bar ], {}, ( x: number ): number => x ); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided a second options argument which is not an object...
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts]:
```suggestion
// The compiler throws an error if the function is provided an options argument which is not an object...
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+{
+	parallel( [ foo, bar ], 2, done ); // $ExpectError
+	parallel( [ foo, bar ], false, done ); // $ExpectError
+	parallel( [ foo, bar ], true, done ); // $ExpectError
+	parallel( [ foo, bar ], 'abc', done ); // $ExpectError
+	parallel( [ foo, bar ], [], done ); // $ExpectError
+	parallel( [ foo, bar ], ( x: number ): number => x, done ); // $ExpectError
+}
+
+// The compiler throws an error if the function is provided a second options argument which is not a supported object...
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts]:
```suggestion
// The compiler throws an error if the function is provided invalid options...
```

File: lib/node_modules/@stdlib/utils/async/parallel/lib/main.js [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+// MODULES //
+
+var factory = require( './factory.js' );
+
+
+// MAIN //
+
+/**
+* Execute functions in parallel and return an array of results.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/async/parallel/lib/main.js]:
```suggestion
* Executes a set of functions in parallel.
```

File: lib/node_modules/@stdlib/utils/async/parallel/lib/index.js [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+/**
+* Execute functions in parallel and return an array of results.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/async/parallel/lib/index.js]:
```suggestion
* Execute a set of functions in parallel.
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/repl.txt [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+    prevent the `done` callback from being invoked.
+
+    If any function calls the provided callback with a truthy `error` argument,
+    the queued functions execution is suspended and immediately calls
+    the completion callback for subsequent error handling.
+
+    Execution is *not* guaranteed to be asynchronous. To ensure asynchrony, wrap
+    the completion callback in a function which either executes at the end of
+    the current stack (e.g., `nextTick`) or during a subsequent turn of the
+    event loop (e.g., `setImmediate`, `setTimeout`).
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/repl.txt]: These should all be wrapped to 80 characters.

File: lib/node_modules/@stdlib/utils/async/parallel/docs/repl.txt [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+    options.thisArg: any (optional)
+        Execution context.
+
+    done: Function
+        Callback to invoke upon completion.
+
+    Returns
+    -------
+    fcn: Function
+        Parallel function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/repl.txt]:
```suggestion
        Function for executing a set of functions in parallel.
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts [hunk 15]
```diff
@@ -0,0 +1,282 @@
...
+	* 	'thisArg': {},
+	* 	'limit': 2
+	* }
+	*
+	* parallel( fcns, opts, done );
+	*/
+	( fcns: Array<Function>, options: Options, clbk: DoneCallback ): void;
+
+	/**
+	* Executes functions in series, passing the results of one function as arguments to the next function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts]:
```suggestion
	* Executes a set of functions in parallel.
```

File: lib/node_modules/@stdlib/utils/async/parallel/package.json [hunk 16]
```diff
@@ -0,0 +1,68 @@
+{
+  "name": "@stdlib/utils/async/parallel",
+  "version": "0.0.0",
+  "description": "Execute functions in parallel and return an array of results."
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/utils/async/parallel/package.json]:
```suggestion
  "description": "Executes a set of functions in parallel."
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts [hunk 17]
```diff
@@ -0,0 +1,282 @@
...
+* Execute functions in parallel, passing the results of all functions to a final callback.
+*/
+type ParallelFunction = () => void;
+
+/**
+* Interface for `parallel`.
+*/
+interface Parallel {
+	/**
+	* Executes functions in series, passing the results of one function as arguments to the next function.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts]:
```suggestion
	* Executes a set of functions in parallel.
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts [hunk 18]
```diff
@@ -0,0 +1,282 @@
...
+	* var fcns = [ foo, bar ];
+	*
+	* var opts = {
+	* 	'thisArg': {},
+	* 	'limit': 2
+	* }
+	*
+	* parallel( fcns, opts, done );
+	*/
+	( fcns: Array<Function>, options: Options, clbk: DoneCallback ): void;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts]:
```suggestion
	( fcns: ArrayLike<Function>, options: Options, clbk: DoneCallback ): void;
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts [hunk 19]
```diff
@@ -0,0 +1,282 @@
...
+	*     }
+	*     console.log( results );
+	*     // => [ 'one', 'two' ]
+	* }
+	*
+	* var fcns = [ foo, bar ];
+	*
+	* parallel( fcns, done );
+	*/
+	( fcns: Array<Function>, clbk: DoneCallback ): void;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts]:
```suggestion
	( fcns: ArrayLike<Function>, clbk: DoneCallback ): void;
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts [hunk 20]
```diff
@@ -0,0 +1,282 @@
...
+	* var opts = {
+	* 	'thisArg': {},
+	* 	'limit': 2
+	* }
+	*
+	* var run = parallel.factory( fcns, opts, done );
+	*
+	* run();
+	*/
+	factory( fcns: Array<Function>, options: Options, clbk: DoneCallback ): ParallelFunction;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts]:
```suggestion
	factory( fcns: ArrayLike<Function>, options: Options, clbk: DoneCallback ): ParallelFunction;
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts [hunk 21]
```diff
@@ -0,0 +1,282 @@
...
+	*     // => [ 2, 4 ]
+	* }
+	*
+	* var fcns = [ a, b ];
+	*
+	* var run = parallel.factory( fcns, done );
+	*
+	* run();
+	*/
+	factory( fcns: Array<Function>, clbk: DoneCallback ): ParallelFunction;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts]:
```suggestion
	factory( fcns: ArrayLike<Function>, clbk: DoneCallback ): ParallelFunction;
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts [hunk 22]
```diff
@@ -0,0 +1,282 @@
...
+*/
+type DoneUnary = ( error: Error ) => void;
+
+/**
+* Callback invoked upon completion.
+*
+* @param error - encountered error or null
+* @param out - output object
+*/
+type DoneBinary = ( error: Error | null, out: Array<unknown> ) => void;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/index.d.ts]: We should be preserving return value type information. E.g.,

```
DoneBinary<T>= ( error: Error | null, out: Array<T> ) => void;
```

File: lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts [hunk 23]
```diff
@@ -0,0 +1,151 @@
...
+* @private 
+* @param error - error argument
+* @param out - output array
+*/
+function done( error: Error | null, out: any  ): void {
+	if ( error ) {
+		throw error;
+	}
+	if ( out === void 0 ) {
+		throw new Error( '`out` is missing.' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/docs/types/test.ts]:
```suggestion
		throw new Error( 'unexpected error' );
```

File: lib/node_modules/@stdlib/utils/async/parallel/lib/limit.js [hunk 24]
```diff
@@ -0,0 +1,134 @@
...
+
+// VARIABLES //
+
+var debug = logger( 'parallel-async:limit' );
+
+
+// MAIN //
+
+/**
+* Invokes the functons in the array, limiting the number of concurrently pending functions.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/lib/limit.js]:
```suggestion
* Invokes functions in a provided array, limiting the number of concurrently pending functions.
```

File: lib/node_modules/@stdlib/utils/async/parallel/README.md [hunk 25]
```diff
@@ -0,0 +1,297 @@
...
+
+```javascript
+var parallel = require( '@stdlib/utils/async/parallel' );
+```
+
+#### parallel( functionArray, \[options,] done )
+
+Executes functions inside `functionArray` in parallel, passing the results of all functions to a final callback.
+
+The first argument to each function is a `resolve` callback. The `resolve` callback should be invoked upon function completion. The first argument is reserved as an `error` argument (which can be `null` or void). If there is a result, it should be provided as the second argument to the `resolve` callback. Failure to invoke the `resolve` callback will prevent the `done` callback from being invoked.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/README.md]: This should be moved to the notes section. There is some duplication here with what is below.

File: lib/node_modules/@stdlib/utils/async/parallel/README.md [hunk 26]
```diff
@@ -0,0 +1,297 @@
...
+    if ( error ) {
+        console.error( error.message );
+    }
+    // => 'beep'
+}
+
+var fcns = [ a, b ];
+
+parallel( fcns, done );
+```
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/README.md]: This appears to be a duplicated example. Am I wrong here?

File: lib/node_modules/@stdlib/utils/async/parallel/README.md [hunk 27]
```diff
@@ -0,0 +1,297 @@
...
+var opts = {
+    'thisArg': ctx
+};
+
+parallel( fcns, opts, done );
+```
+
+#### parallel.factory( functionArray, \[options,] done )
+
+Returns a reusable parallel `function`.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/README.md]:
```suggestion
Returns a reusable parallel function.
```

File: lib/node_modules/@stdlib/utils/async/parallel/README.md [hunk 28]
```diff
@@ -0,0 +1,297 @@
...
+
+<!-- /.usage -->
+
+<!-- Package usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="notes">
+
+## Notes
+
+-   The `functionArray` argument should be an array of functions to be executed in parallel.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/utils/async/parallel/package.json [hunk 29]
```diff
@@ -0,0 +1,68 @@
+{
+  "name": "@stdlib/utils/async/parallel",
+  "version": "0.0.0",
+  "description": "Executes a set of functions in parallel."
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/package.json]:
```suggestion
  "description": "Execute a set of functions in parallel."
```

File: lib/node_modules/@stdlib/utils/async/parallel/test/test.main.js [hunk 30]
```diff
@@ -0,0 +1,447 @@
...
+		3
+	];
+
+	function done( error, out ) {
+		if ( error ) {
+			t.fail( error.message );
+		} else {
+			t.pass( 'did not return an error' );
+		}
+		t.deepEqual( out, expected, 'returns expected output' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/test/test.main.js]:
```suggestion
		t.deepEqual( out, expected, 'returns expected value' );
```

Here and below. We try to be consistent in our assertion messages.

File: lib/node_modules/@stdlib/utils/async/parallel/test/test.main.js [hunk 31]
```diff
@@ -0,0 +1,447 @@
...
+		t.end();
+	}
+});
+
+tape( 'The function invokes the provided functions once and return the results in order (synchronous)', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/test/test.main.js]:
```suggestion
tape( 'the function invokes the provided functions once and return the results in order (synchronous)', function test( t ) {
```

Always lowercase the first letter of test descriptions here and elsewhere.

File: lib/node_modules/@stdlib/utils/async/parallel/test/test.main.js [hunk 32]
```diff
@@ -0,0 +1,447 @@
...
+		if ( error ) {
+			t.fail( error.message );
+		} else {
+			t.pass( 'did not return an error' );
+		}
+		t.end();
+	}
+});
+
+tape( 'if an error is encountered while processing an object property, The function suspends execution and immediately returns the error (limit = 1)', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/test/test.main.js]:
```suggestion
tape( 'if an error is encountered while processing an object property, the function suspends execution and immediately returns the error (limit = 1)', function test( t ) {
```

File: lib/node_modules/@stdlib/utils/async/parallel/test/test.main.js [hunk 33]
```diff
@@ -0,0 +1,447 @@
...
+		if ( error ) {
+			t.pass( error.message );
+		} else {
+			t.fail( 'did not return an error' );
+		}
+		t.end();
+	}
+});
+
+tape( 'if an error is encountered while processing an object property, The function suspends execution and immediately returns the error (concurrent; limit)', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/test/test.main.js]:
```suggestion
tape( 'if an error is encountered while processing an object property, the function suspends execution and immediately returns the error (concurrent; limit)', function test( t ) {
```

Comment (kgryte) [reply lib/node_modules/@stdlib/utils/async/parallel/test/test.main.js]: Here and elsewhere.

File: lib/node_modules/@stdlib/utils/async/parallel/test/test.factory.js [hunk 34]
```diff
@@ -0,0 +1,478 @@
...
+	t.end();
+});
+
+tape( 'the function returns a function (options)', function test( t ) {
+	var f = factory( [noop], {}, noop );
+	t.equal( typeof f, 'function', 'returns a function' );
+	t.end();
+});
+
+tape( 'the returned function invokes the provided functions once and return the results in order (asynchronous)', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/test/test.factory.js]:
```suggestion
tape( 'the returned function invokes the provided functions once and returns the results in order (asynchronous)', function test( t ) {
```

File: lib/node_modules/@stdlib/utils/async/parallel/test/test.factory.js [hunk 35]
```diff
@@ -0,0 +1,478 @@
...
+			t.fail( error.message );
+		} else {
+			t.pass( 'did not return an error' );
+		}
+		t.deepEqual( out, expected, 'returns expected output' );
+		t.end();
+	}
+});
+
+tape( 'the returned function invokes the provided functions once and return the results in order (synchronous)', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/async/parallel/test/test.factory.js]:
```suggestion
tape( 'the returned function invokes the provided functions once and returns the results in order (synchronous)', function test( t ) {
```

Here and elsewhere.

File: lib/node_modules/@stdlib/utils/async/parallel/README.md (hunk 26 above)
Comment (nightknighto) [reply lib/node_modules/@stdlib/utils/async/parallel/README.md]: @kgryte This example showcases handling errors. Should it be removed?

Comment (kgryte) [reply lib/node_modules/@stdlib/utils/async/parallel/README.md]: I think we can remove. We typically don't showcase exception behavior in README examples.

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Deadreyo.

I ended up refactoring a bit. We seem to diverge somewhat across `@stdlib/utils/async` in terms of whether the function returned by a `factory` method should accept a `done` argument. I ended up changing the implementation to require `done` be provided to the returned function. This made a bit more sense in my mind in terms of how I'd expect the reusable function to be used. One could probably make the argument that we should update `series-waterfall` similarly.

**Review by kgryte — APPROVED** [review]

**Pull request merged**